saleor/checkout/complete_checkout.py

```diff
@@ -355,6 +352,15 @@ def _create_order_line_discounts(
     return line_discounts
 
 
+def _get_sale_id(line_discounts: list[OrderLineDiscount]):
+    sale_id = None
+    for discount in line_discounts:
+        if discount.type == DiscountType.PROMOTION:
+            if rule := discount.promotion_rule:
+                sale_id = get_sale_id(rule.promotion)
+    return sale_id
```

Comment:
from above usage, I understand this is meant to return one possible sale, maybe we can simplify it?
```suggestion
def _get_sale_id(line_discounts: list[OrderLineDiscount]):
    for discount in line_discounts:
        if discount.type == DiscountType.PROMOTION:
            if rule := discount.promotion_rule:
                return get_sale_id(rule.promotion)
```